web/app/room/[roomId]/page.jsx

```
@@ -369,26 +649,314 @@ export default function RoomPage({ params }) {
+      {flagging && (
+        <div className="fixed inset-0 bg-black/50 flex items-center justify-center p-4 z-50">
+          <Card className="w-full max-w-md">
+            <CardContent className="p-6 space-y-4">
+              <h3 className="text-lg font-semibold">Flag Submission</h3>
+              <p className="text-xs font-mono text-muted-foreground break-all">
+                {flagging.submission_id}
+              </p>
+              <form onSubmit={onFlagSubmit} className="space-y-4">
+                <div>
+                  <label className="text-sm font-medium">Reason</label>
+                  <textarea
```

Biome (2.1.2)

[error] A form label must be associated with an input.

Consider adding a `for` or `htmlFor` attribute to the label element or moving the input element to inside the label element.

(lint/a11y/noLabelWithoutControl)